pkg/epp/handlers/streamingserver.go

```diff
@@ -364,9 +366,43 @@ func (s *StreamingServer) HandleRequestBody(
 		return reqCtx, errutil.Error{Code: errutil.Internal, Msg: fmt.Sprintf("error marshaling request body: %v", err)}
 	}
 
-	target, err := s.scheduler.Schedule(ctx, llmReq)
+	schedulableReq, err := newSchedulableRequestFromContext(reqCtx)
 	if err != nil {
-		return reqCtx, errutil.Error{Code: errutil.InferencePoolResourceExhausted, Msg: fmt.Errorf("failed to find target pod: %w", err).Error()}
+		return nil, errutil.Error{Code: errutil.Internal, Msg: err.Error()}
+	}
+	target, evictionReason, err := s.queueController.Schedule(schedulableReq)
+	if err != nil {
+		logger.Error(err, "Failed to schedule request", "evictionReason", evictionReason.String())
+		switch {
+		case errors.Is(err, scheduling.ErrEvicted):
```

Comment:
should be encapsulated in a function

Reply:
sounds good; I can also do the encapsulate the API error conversion inside the QueueController, but I thought it would be cleaner to have the QueueController as a library with clear error contracts and the streamingserver handle all the API stuff

Reply:
Note that this file is now gone from upstream, the non-streaming path was removed, and the streaming path replaced it in request.go/response.go/server.go